lte/gateway/c/core/oai/tasks/amf/nas_proc.cpp

```diff
@@ -883,21 +882,23 @@ int amf_decrypt_imsi_info_answer(itti_amf_decrypted_imsi_info_ans_t* aia) {
                                (supi_imsi.plmn.mnc_digit3 & 0xf);
   }
 
+  imsi64 = amf_imsi_to_imsi64(params->imsi);
+  ue_context->amf_context.imsi64 = imsi64;
+
   amf_app_generate_guti_on_supi(&amf_guti, &supi_imsi);
   amf_ue_context_on_new_guti(ue_context,
                              reinterpret_cast<guti_m5_t*>(&amf_guti));
 
   ue_context->amf_context.m5_guti.m_tmsi = amf_guti.m_tmsi;
   ue_context->amf_context.m5_guti.guamfi = amf_guti.guamfi;
-  imsi64 = amf_imsi_to_imsi64(params->imsi);
 
   params->decode_status = ue_context->amf_context.decode_status;
   /*
    * Execute the requested new UE registration procedure
    * This will initiate identity req in DL.
    */
-  rc = amf_proc_registration_request(aia->ue_id, is_amf_ctx_new, params);
-  OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
+  amf_proc_registration_request(aia->ue_id, is_amf_ctx_new, params);
+  OAILOG_FUNC_RETURN(LOG_NAS_AMF, imsi64);
```

Comment:
Could change the return value at line, https://github.com/magma/magma/pull/12088/files#diff-359e73baac363f6d67312a481dde16bd3d45d72538d19589bb94f57e5ac29385R842

Reply:
addressed